internal/models/user.go

```diff
@@ -528,7 +529,9 @@ func IsDuplicatedEmail(tx *storage.Connection, email, aud string) (*User, error)
 	userIDs := make(map[string]uuid.UUID)
 	for _, identity := range identities {
 		if !identity.IsForSSOProvider() {
-			userIDs[identity.UserID.String()] = identity.UserID
+			if (currentUser != nil && currentUser.ID != identity.UserID) && (currentUser == nil) {
```

Comment:
this is where the main change is made

Reply:
Haven't looked deeply but should this be an or? Looks like current statement will always return false bc `currentUser != nil` and `currentUser == nil` can't be true concurrently
```suggestion 
			if (currentUser != nil && currentUser.ID != identity.UserID) || (currentUser == nil) {
```

Reply:
Should this be an or statement? Haven't looked deeply but seems like `currentUser != nil` and 
`currentUser == nil` can't concurrently be true

```suggestion
			if (currentUser != nil && currentUser.ID != identity.UserID) ||(currentUser == nil) {
```